internal/controllers/machinehealthcheck/machinehealthcheck_targets.go

```diff
@@ -129,12 +120,99 @@ func (t *healthCheckTarget) needsRemediation(logger logr.Logger, timeoutForMachi
 		return false, 0
 	}
 
+	// Collect all unhealthy conditions (both node and machine) to provide comprehensive status
+	// Always evaluate machine conditions regardless of node state for consistent behavior
+	var (
+		unhealthyNodeMessages    []string
+		unhealthyMachineMessages []string
+	)
+
+	// Always check machine conditions first, regardless of node state
+	for _, c := range t.MHC.Spec.Checks.UnhealthyMachineConditions {
+		machineCondition := getMachineCondition(t.Machine, c.Type)
+
+		// Skip when current machine condition is different from the one reported
+		// in the MachineHealthCheck.
+		if machineCondition == nil || machineCondition.Status != c.Status {
+			continue
+		}
+
+		// If the machine condition has been in the unhealthy state for longer than the
+		// timeout, mark as unhealthy and collect the message.
+		timeoutSecondsDuration := time.Duration(ptr.Deref(c.TimeoutSeconds, 0)) * time.Second
+
+		if machineCondition.LastTransitionTime.Add(timeoutSecondsDuration).Before(now) {
+			unhealthyMachineMessages = append(unhealthyMachineMessages, fmt.Sprintf("Condition %s on Machine is reporting status %s for more than %s", c.Type, c.Status, timeoutSecondsDuration.String()))
+			logger.V(3).Info("Target is unhealthy: machine condition is in state longer than allowed timeout", "condition", c.Type, "state", c.Status, "timeout", timeoutSecondsDuration.String())
+			continue
+		}
+
+		durationUnhealthy := now.Sub(machineCondition.LastTransitionTime.Time)
+		nextCheck := timeoutSecondsDuration - durationUnhealthy + time.Second
+		if nextCheck > 0 {
+			nextCheckTimes = append(nextCheckTimes, nextCheck)
+		}
+	}
+
+	// Machine has Status.NodeRef set, although we couldn't find the node in the workload cluster.
+	if t.nodeMissing {
+		logger.V(3).Info("Target is unhealthy: node is missing")
+
+		// Always merge node missing message with any unhealthy machine conditions
+		nodeMissingMessage := fmt.Sprintf("Node %s has been deleted", t.Machine.Status.NodeRef.Name)
+		allMessages := append([]string{nodeMissingMessage}, unhealthyMachineMessages...)
+
+		reason := clusterv1.MachineHealthCheckNodeDeletedReason
+		v1beta1Reason := clusterv1.NodeNotFoundV1Beta1Reason
+		if len(unhealthyMachineMessages) > 0 {
+			reason = clusterv1.MachineHealthCheckUnhealthyMachineReason
+			v1beta1Reason = clusterv1.UnhealthyMachineConditionV1Beta1Reason
+		}
+
+		// For v1beta2 we use a single-line message prefixed with "Health check failed: "
```

Comment:
For v1beta2 should use multiline comment with * prefix on every message (I pasted an example of how to compute it in my previous comment)

 Also, we usually avoid to call out v1beta2 (v1beta2 conditions are just conditions), so the  we won't need to fix comment when v1beta1 code will go away

(Same for other places where we are computing condition message)

Reply:
Removed v1beta2 mentions and fixed the the comment to use a multiline comment with * prefix now.